warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/BaseIngestHelper.java

```diff
@@ -603,6 +603,10 @@ public static Matcher compileFieldNamePattern(String fieldNamePattern) {
         
         if (types.isEmpty()) {
             types.addAll(typeFieldMap.get(null));
+        } else {
+            synchronized (this) {
+                typeFieldMap.putAll(typeFieldName, types);
```

Comment:
It's probably not a huge deal, but whenever the typeFieldMap already contains an entry for typeFieldName, we'll re-add the types list. `putAll` on a multimap iterates over the entire collection you are adding. I would probably be better to only insert into the map if we never found it there in the first place.

Reply:
I think I would have to say that 
1) we do not need to synchronize.  If we do then we need to add synchronization in several other places
2) instead of an else clause here, simply change line 599 as follows:
     Collection<datawave.data.type.Type<?>> patternTypes = typeCompiledPatternMap.get(patternMatcher);
     types.addAll(patternTypes);
     typeFieldMap.putAll(typeFieldName, patternTypes);